src/CodeGen_Internal.cpp

```diff
@@ -380,7 +380,7 @@ Expr lower_int_uint_mod(const Expr &a, const Expr &b) {
 
     int bits;
     if (is_const_power_of_two_integer(b, &bits)) {
-        return a & (b - 1);
+        return a & simplify(b - 1);
```

Comment:
If we're gonna call simplify() here, why not do the entire expression?

Reply:
Excessive simplification is bad, both for performance, and because it replaces intrinsics with "simpler" arithmetic sometimes. In this case, simplify is guaranteed to just be constant folding (b is constant, b - 1 should constant fold).